### DISCUSSION TITLE
Storing many images in Next.JS

### DISCUSSION THREAD

**Discussion by BearCooder** [vercel/next.js#55485] [Help] [answered]
### Summary

Hi, so there is one thing that is not clear or at least a bit confusing to me.

I have a Gallery website that displays multiple images(in total over 150) in high resolution. So all images together end up big in size.

What I currently did in development is, I stored all images in a dedicated folder in my Next.js project - in `src/media/`. 

And in `src/app/artworks/page.tsx ` I display all images. Now this page file is a server component.

If I understand correctly this way should be fine as it is a server component and the images stored in the media folder do not add to the bundle size and are rendered and cached on the server?

Is my understanding correct? Or should I not store all the images in my next js project and store them in a database somewhere else?

### Example

https://codesandbox.io/p/sandbox/eager-mclean-xh5qjh?file=/app/gallery/[id]/page.tsx:1,1

**Comment by stevegiotis** [comment, marked as answer]
hey mate, your approach of storing images in a dedicated folder within your Next.js project is reasonable, especially if you don't want the complexity of managing images in a database. However, ensure that you optimize your images and consider implementing lazy loading to improve the initial page load performance, especially when dealing with a large number of high-resolution images.

**Reply by BearCooder** [reply]
Hey @stevegiotis , thank you very much for your answer. Yes I plan to implement lazy loading and all the images are high resolution but already .webp files and I use the Image component from Next.js. If you have any other tips please feel free to let me know. Happy to learn!
